### PR TITLE
Rework index.md and create separate documents to explain RAM, bandwidth, etc 

### PR DESCRIPTION
**Pull request by iamveritas**
## Change Description
1. Spliting index.md file, the content thus moved into new sections, index.md has link to those new sections
2. Fix outdated links
3. Small corrections on emphasizing terms, e.g. `system contracts` or `system accounts`
5. Some re-ordering of the sections was done due to introductions of the new ones
6. No new content added, thus another tech review or content review is not needed; however annotations of the system contract's classes were added or updated.
7. To avoid double listing of system contracts (in `reference doc tree` and under `core concepts tree`) on the main navigation, each system contract class was annotated more thoroughly; these annotations will make their way into the `reference doc` content. The core concepts subsection in the navigation tree is not needed anymore, core concept are covered by index.md with link to the reference doc

## Deployment Changes
- [ ] Deployment Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x] Documentation Additions

### CODE REVIEW

File: docs/02_system_contracts/05_eosio_wrap.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## eosio.wrap system contract
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/05_eosio_wrap.md]: I will change to 
[eosio.wrap system contract](action-reference/eosio.wrap)

File: docs/02_system_contracts/04_eosio_token.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## eosio.token system contract
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/04_eosio_token.md]: same approach as above

File: docs/02_system_contracts/03_eosio_msig.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+## eosio.msig system contract
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/03_eosio_msig.md]: same approach as above

File: docs/02_system_contracts/01_eosio_bios.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+## eosio.bios system contract
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/01_eosio_bios.md]: same approach as above

File: docs/02_system_contracts/01_eosio_bios.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+The `eosio.bios` is the first sample of system smart contract provided by `block.one` through the EOSIO platform. It is a minimalist system contract because it only supplies the actions that are absolutely critical to bootstrap a chain and nothing more. This allows for a chain agnostic approach to bootstrapping a chain.
+
+The actions implemented and publicly exposed by `eosio.bios` system contract are: setpriv, setalimits, setglimits, setprods, setparams, reqauth, setabi.
+
+|Action name|Action description|
+|---|---|
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/01_eosio_bios.md]: I will take out the details about each action, and then point to the reference documentation for details, however we have to list them cause this contract is a bit special, some of them actions are not implemented at this level, they are just empty actions that just map to the chain implementation. This distinction, which is tricky to get and understand is not revealed in reference documentation. And in order to explain it here, we have to list the actions that have an implementation and those that don't. 
In short, we will end up with a list of actions but their details will be in reference documentation only.

File: docs/02_system_contracts/02_eosio_system.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+- Users can stake tokens for CPU and Network bandwidth, and then vote for producers or delegate their vote to a proxy.
+- Producers can register in order to be voted for, and can claim per-block and per-vote rewards.
+- Users can buy and sell RAM at a market-determined price.
+- Users can bid on premium names.
+- A resource exchange system, named REX, allows token holders to lend their tokens, and users to rent CPU and NET resources in return for a market-determined fee.
+
+The actions implemented and publicly exposed by the `eosio.system` system contract are presented in the table below. Just like the `eosio.bios` sample contract there are a few actions which are not implemented at the contract level (`newaccount`, `updateauth`, `deleteauth`, `linkauth`, `unlinkauth`, `canceldelay`, `onerror`, `setabi`, `setcode`), they are just declared in the contract so they will show in the contract's ABI and users will be able to push those actions to the chain via the account holding the 'eosio.system' contract, but the implementation is at the EOSIO core level. They are referred to as EOSIO native actions.
+
+|Action name|Action description|
+|---|---|
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/02_eosio_system.md]: I pruned the duplicated content.

File: docs/02_system_contracts/03_eosio_msig.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+The workflow to propose, review, approve and then executed a transaction is describe in details [here](../04_guides/06_how-to-sign-a-multisig-transaction-with-eosio.msig.md), and in short it can be described by the following:
+- first you create a transaction json file, 
+- then you submit this proposal to the `eosio.msig` contract, and you also insert the account permissions required to approve this proposal into the command that submits the proposal to the blockchain,
+- the proposal then gets stored on the blockchain by the `eosio.msig` contract, and is accessible for review and approval to those accounts required to approve it,
+- after each of the appointed accounts required to approve the proposed transactions reviews and approves it, you can execute the proposed transaction.  The `eosio.msig` contract will execute it automatically, but not before validating that the transaction has not expired, it is not cancelled, and it has been signed by all the permissions in the initial proposal's required permission list.
+
+These are the actions implemented and publicly exposed by the `eosio.msig` contract:
+|Action name|Action description|
+|---|---|
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/03_eosio_msig.md]: I pruned the duplicated content.

File: docs/02_system_contracts/04_eosio_token.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+The `eosio.token` contract class also implements two useful public static methods: `get_supply` and `get_balance`. The first allows one to check the total supply of a specified token, created by an account and the second allows one to check the balance of a token for a specified account (the token creator account has to be specified as well).
+
+The `eosio.token` contract manages the set of tokens, accounts and their corresponding balances, by using two internal multi-index structures: the `accounts` and `stats`. The `accounts` multi-index table holds, for each row, instances of `account` object and the `account` object holds information about the balance of one token. If we remember how multi-index tables work, see [here](https://developers.eos.io/eosio-cpp/docs/using-multi-index-tables), then we understand also that the `accounts` table is scoped to an eosio account, and it keeps the rows indexed based on the token's symbol.  This means that when one queries the `accounts` multi-index table for an account name the result is all the tokens that account holds at the moment.
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/04_eosio_token.md]: also corrected in a couple of other places

File: docs/index.md [hunk 9]
```diff
@@ -4,208 +4,22 @@ The EOSIO blockchain platform is unique in that the features and characteristics
...
+1. [eosio.bios](02_system_contracts/01_eosio_bios.md)
+2. [eosio.system](02_system_contracts/02_eosio_system.md)
+3. [eosio.msig](02_system_contracts/03_eosio_msig.md)
+4. [eosio.token](02_system_contracts/04_eosio_token.md)
+5. [eosio.wrap](02_system_contracts/05_eosio_wrap.md)
 
-Why is it easier for governance actions to be executed via this contract?
-The answer to this question is explained in detailed [here](./03_guides/07_how-to-use-eosio.wrap.md)
\ No newline at end of file
+## Build and deploy
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/index.md]: build and deploy it is on the main navigation as well, in plain sight, not hidden.

File: docs/02_system_contracts/04_eosio_token.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/04_eosio_token.md]: removed

File: docs/02_system_contracts/01_eosio_bios.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+Below are listed the actions which are declared in the `eosio.bios` contract, mapped one-to-one with the native EOSIO actions, but having no implementation at the contract level:
+
+|Action name|Description|
```
[the remark this answers is not in the document]
Comment (iamveritas) [reply docs/02_system_contracts/01_eosio_bios.md]: the intention here is to list separately the actions which are implemented at the contract level vs the action which are only declared at the contract level and implemented at the core level.
I have pruned the duplicated content, however the two lists of the actions which clearly separates them based on the above property I left in. This particular aspect is tricky to get when you first encounter the eosio.bios contract.
